src/components/SearchBar.vue

```diff
@@ -0,0 +1,103 @@
+<template>
+  <div>
+    <oc-search-bar
+      ref="globalSearchBar"
+      class="uk-width-large uk-visible@m"
+      :label="searchLabel"
+      :loading="isLoadingSearchResults"
+      :button-hidden="true"
+      @search="search"
+      @clear="resetSearch"
+    />
+    <oc-button
+      id="files-open-search-btn"
+      key="mobile-search-button"
+      class="uk-hidden@m"
+      icon="search"
+      aria-label="search"
+      @click="focusMobileSearchInput"
+    />
+    <oc-drop
+      drop-id="oc-topbar-search-mobile"
+      toggle="#files-open-search-btn"
+      mode="click"
+      class="uk-margin-remove"
+      :options="{ delayHide: '0' }"
+    >
+      <oc-search-bar
+        ref="mobileSearch"
+        :label="searchLabel"
+        :loading="isLoadingSearchResults"
+        @search="search"
+        @clear="resetSearch"
+      />
+    </oc-drop>
+  </div>
+</template>
+
+<script>
+import { mapActions } from 'vuex'
+
+export default {
+  name: 'SearchBar',
+
+  data: () => ({
+    isLoadingSearchResults: false
+  }),
+
+  computed: {
+    searchLabel() {
+      return this.$gettext('Search')
+    }
+  },
+
+  watch: {
+    $route() {
+      this.clearSearchBarValue()
+    }
+  },
+
+  methods: {
+    ...mapActions('Files', ['searchForFile', 'resetSearch']),
+
+    search(searchTerm) {
+      if (!searchTerm) {
+        return
```

Comment:
Since typing `enter` emits `search`, it would make sense to call `resetSearch` here just above the return.